src/utils/codeBuilder.ts

```diff
@@ -0,0 +1,309 @@
+    static readonly TEMPLATE_VARIABLES = 'TEMPLATE_VARIABLES';
+    static readonly TEMPLATE_VARIABLE_ASSIGNMENTS =
+        'TEMPLATE_VARIABLE_ASSIGNMENTS';
+    static readonly TEMPLATE_VIEW_LWC_LABEL = 'TEMPLATE_VIEW_LWC_LABEL';
+    static readonly TEMPLATE_LWC_NAME = 'TEMPLATE_LWC_NAME';
+    static readonly TEMPLATE_QUICK_ACTION_ICON = 'TEMPLATE_QUICK_ACTION_ICON';
+    static readonly TEMPLATE_QUICK_ACTION_LABEL = 'TEMPLATE_QUICK_ACTION_LABEL';
+
+    private extensionUri: Uri;
+    private objectApiName: string;
+    templateVariables: TemplateVariables;
+    fieldNames: string[];
+
+    constructor(
+        extensionUri: Uri,
+        objectApiName: string,
+        compactLayoutFields: CompactLayoutField[]
+    ) {
+        this.extensionUri = extensionUri;
+        this.objectApiName = objectApiName;
+
+        this.fieldNames = this.getFieldNames(compactLayoutFields);
+        this.templateVariables = this.generateTemplateVariables(
+            this.fieldNames
+        );
+    }
+
+    async generateView(): Promise<boolean> {
+        return new Promise(async (resolve) => {
+            const lwcName = `view${this.objectApiName}Record`;
+            this.copyTemplateFiles(
+                this.templateVariables,
+                'viewRecord',
+                lwcName
+            );
+            this.createQuickAction(this.templateVariables, 'View', lwcName);
+            resolve(true);
+        });
+    }
+
+    async generateEdit(): Promise<boolean> {
+        return new Promise(async (resolve) => {
+            const lwcName = `edit${this.objectApiName}Record`;
+            this.copyTemplateFiles(
+                this.templateVariables,
+                'editRecord',
+                lwcName
+            );
+            this.createQuickAction(
+                this.templateVariables,
+                'Edit',
+                lwcName,
+                'editActionIcon'
+            );
+            resolve(true);
+        });
+    }
+
+    async generateCreate(): Promise<boolean> {
+        return new Promise(async (resolve) => {
+            const lwcName = `create${this.objectApiName}Record`;
+            this.copyTemplateFiles(
+                this.templateVariables,
+                'createRecord',
+                lwcName
+            );
+            this.createQuickAction(this.templateVariables, 'Create', lwcName);
+            resolve(true);
+        });
+    }
+
+    private getFieldNames(compactLayoutFields: CompactLayoutField[]) {
+        const fieldNames: string[] = [];
+        compactLayoutFields.forEach((field) => {
+            field.layoutComponents.forEach((component) => {
+                fieldNames.push(component.value);
+            });
+        });
+        return fieldNames;
+    }
+
+    private createQuickAction(
+        templateVariables: TemplateVariables,
+        label: string,
+        name: string,
+        iconName: string | undefined = undefined
```

Comment:
Nit: optional properties will default to `undefined`, if they're not passed in. You don't need the default assignment.

Reply:
I am getting a compile error if I dont specify it from the invocations above.

Reply:
Ah, it's because of the specific nomenclature, which I always get wrong: for optional method parameters specifically, you have to declare the optional input parameter as `iconName?: string`. This is distinct from declaring a variable as type `<type> | undefined` (e.g. `let myOptionalString: string | undefined;`. The fact that these feel like a distinction without a difference is what always makes me forget which one to use where.

Anyway, long story short: the proper way to code this optional parameter in a method, in TypeScript, is to declare it as `iconName?: string`.

Reply:
Aha! Updated!